hal/common/critical.c

```diff
@@ -270,7 +270,7 @@ uint32_t core_util_atomic_incr_u32(uint32_t *valuePtr, uint32_t delta)
 }
 
 void *core_util_atomic_incr_ptr(void **valuePtr, ptrdiff_t delta) {
-    return core_util_atomic_incr((uintptr_t)valuePtr, (uintptr_t)delta);
+    return core_util_atomic_incr_u32((uintptr_t)valuePtr, (uintptr_t)delta);
```

Comment:
Ah ok, here's a quick example with a simple stack:
``` c
void stack_push(struct stack *s, uint16_t data) {
    uint16_t *reserved = core_util_atomic_decr_ptr(&s->pos, sizeof(uint16_t));
    *reserved = data; // note, an additional fetch may be intervened by another decr
}
```

Although the uses for atomic_incr/decr_ptr are quite limited since most algorithms involve extra work on the pointer to account for array boundaries. I added the atomci_*_ptr versions for consistency with the other types, but I'm not opposed to dropping them.